### PR TITLE
[Feat] Add DRAM pool KV protocol with unified pack/unpack

### PR DESCRIPTION
**Pull request by yumingyue624**
## Purpose
Implement pack/unpack for the DRAM pool KV transport (send + flag buffer), mirroring the ASU architecture.

## Modifications
1. common: shared types, constants, and helpers (PeekOpcode, VerifyHeader, PackHeader, ValidateRequestHeader).
2. client: KvDumpLoadProtocol/KvLookupProtocol + ProtocolManager for PackRequest and UnpackResponse.
3. server: KvDumpLoadServerProtocol/KvLookupServerProtocol + ServerProtocolManager for UnpackRequest (with inline validation) and PackResponse.
4. CMakeLists: drampool_kv_common/client/server lib targets.
5. Tests: 29 cases covering pack/unpack round-trips, boundary values, multi-entry, opcode validation, and edge cases.

## Test
- drampool.test: 29 gtests passing on x64 (MSVC) and aarch64 (GCC 12.3.1).

### CODE REVIEW

File: ucm/transport/kv/drampool/trans/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#Shared wire - format types, constants and helpers.Linked by both client and server.
+add_library(drampool_kv_common STATIC src / kv_protocol_common.cpp) set_target_properties(
```
Comment (yuanzhg078) [inline ucm/transport/kv/drampool/trans/CMakeLists.txt]: Source file paths in CMake must not contain spaces.

File: ucm/transport/kv/drampool/trans/src/kv_protocol_client.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    std::size_t PackedSize(const KvRequest& req) const override;
+    Status PackRequest(const KvRequest& req, void* target) override;
+    Status UnpackResponse(const void* data, std::uint16_t result_count,
+                          KvResponse& out) const override;
+
+private:
+    Status ValidateRequest(const KvLookupRequest& req) const;
+};
+
+class ProtocolManager {
```
Comment (Infinite666) [inline ucm/transport/kv/drampool/trans/src/kv_protocol_client.h]: I think it's ok to merge client and server together as one parser class.

File: ucm/transport/kv/drampool/trans/src/kv_protocol_server.cpp [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+#include <cstring>
+#include <string>
+
+namespace UC::DRAMPOOL {
+
+Status KvDumpLoadServerProtocol::UnpackRequest(const void* data, std::size_t size,
+                                               std::unique_ptr<KvRequest>& out) const
+{
+    if (!data) {
+        return Status::Error(StatusCode::INVALID_ARGUMENT, "DumpLoad: UnpackRequest data is null");
```
Comment (yuanzhg078) [inline ucm/transport/kv/drampool/trans/src/kv_protocol_server.cpp]: The DumpLoadServerProtocol::UnpackRequest header checks here overlap with **VerifyHeader**, but are currently implemented inline instead of reusing the helper.

File: ucm/transport/kv/drampool/trans/CMakeLists.txt (hunk 1 above)
Comment (yumingyue624) [reply ucm/transport/kv/drampool/trans/CMakeLists.txt]: This CMake file was a temporary artifact and has been removed

File: ucm/transport/kv/drampool/trans/src/kv_protocol_client.h (hunk 2 above)
Comment (yumingyue624) [reply ucm/transport/kv/drampool/trans/src/kv_protocol_client.h]: Done. Merged the client and server into a single parser class as suggested.

File: ucm/transport/kv/drampool/trans/src/kv_protocol_server.cpp (hunk 3 above)
Comment (yumingyue624) [reply ucm/transport/kv/drampool/trans/src/kv_protocol_server.cpp]: Done. Removed the redundant inline header checks in DumpLoadServerProtocol::UnpackRequest and now reuse the VerifyHeader helper instead.
